internal/datastore/proxy/mapping.go

```diff
@@ -228,23 +228,46 @@ func (mp mappingProxy) QueryTuples(
 	return &mappingTupleIterator{rawIter, mp.mapper, nil}, nil
 }
 
-func (mp mappingProxy) ReverseQueryTuplesFromSubject(subject *v0.ObjectAndRelation, revision datastore.Revision) datastore.ReverseTupleQuery {
-	translatedONR, err := translateONR(subject, mp.mapper.Encode)
-	return mappingReverseTupleQuery{mp.delegate.ReverseQueryTuplesFromSubject(translatedONR, revision), mp.mapper, err}
-}
+func (mp mappingProxy) ReverseQueryTuples(
+	ctx context.Context,
+	filter *v1.SubjectFilter,
+	revision datastore.Revision,
+	opts ...options.ReverseQueryOptionsOption,
+) (datastore.TupleIterator, error) {
+	subjectType, err := mp.mapper.Encode(filter.SubjectType)
+	if err != nil {
+		return nil, fmt.Errorf(errTranslation, err)
+	}
 
-func (mp mappingProxy) ReverseQueryTuplesFromSubjectNamespace(subjectNamespace string, revision datastore.Revision) datastore.ReverseTupleQuery {
-	translatedNamespace, err := mp.mapper.Encode(subjectNamespace)
-	return mappingReverseTupleQuery{mp.delegate.ReverseQueryTuplesFromSubjectNamespace(translatedNamespace, revision), mp.mapper, err}
-}
+	queryOpts := options.NewReverseQueryOptionsWithOptions(opts...)
 
-func (mp mappingProxy) ReverseQueryTuplesFromSubjectRelation(subjectNamespace, subjectRelation string, revision datastore.Revision) datastore.ReverseTupleQuery {
-	translatedNamespace, err := mp.mapper.Encode(subjectNamespace)
-	return mappingReverseTupleQuery{
-		mp.delegate.ReverseQueryTuplesFromSubjectRelation(translatedNamespace, subjectRelation, revision),
-		mp.mapper,
-		err,
+	translatedOptions := []options.ReverseQueryOptionsOption{
+		options.WithReverseLimit(queryOpts.ReverseLimit),
 	}
+	if queryOpts.ResRelation != nil {
```

Comment:
Same check here to ensure we aren't use any other options than those handled already